src/cloudant/account.py

```diff
@@ -392,32 +392,38 @@ def _usage_endpoint(self, endpoint, year=None, month=None):
         optional year and month URL elements.
 
         :param str endpoint: Cloudant usage endpoint.
-        :param int year: Year to query against.  Defaults to None.
-        :param int month: Month to query against.  Defaults to None.
-        """
-        if year is not None:
-            endpoint = posixpath.join(endpoint, str(year))
-        if month is not None:
-            if year is None:
-                raise CloudantException(
-                    (
-                        "must supply both year and month "
-                        "to usage endpoint: {0}"
-                    ).format(endpoint)
-                )
-            endpoint = posixpath.join(endpoint, str(month))
-        resp = self.r_session.get(endpoint)
+        :param int year: Year to query against.  Optional parameter.
+            Defaults to None.  If used, it must be accompanied by ``month``.
+        :param int month: Month to query against that must be an integer
+            between 1 and 12. Optional parameter. Defaults to None.
+            If used, it must be accompanied by ``year``.
+        """
+        if year and month in range(1, 13):
```

Comment:
I think the trick here is to implement validation that ensures that the message described (here)[https://github.com/cloudant/python-cloudant/pull/120/files/be827cbda13b61a1024377db42807ecfd5fac058#r57162205] works.

What I mean is that if the year, for example, is a string of '2016', it should not be rejected by this validation because 2016 is a valid year but if the year is 'foo' then it should not pass the validation.  I think the validation logic can be implemented so that the message: `'Invalid year and/or month supplied.  Found: year - {0}, month - {1}'.format(year, month)` works for any condition.